qemu/tests/migration_multi_host_with_xbzrle.py

```diff
@@ -0,0 +1,363 @@
+import logging
+from autotest.client.shared import error
+from virttest import utils_test
+from virttest import virt_vm
+from qemu.tests.MigrationMultiHost import MigrationBase
+
+
+@error.context_aware
+def run(test, params, env):
+
+    """
+    KVM multi-host migration test:
+
+    Migration execution progress is described in documentation
+    for migrate method in class MultihostMigration.
+
+    The test procedure:
+    1) starts vm on master host.
+    2) query migrate capabilities, query cache size, enable/disable xbzrle
+    3) a. Migrate guest with heavy loaded inside guest (running a dirty page
+       generator) and compare the total time, downtime and transferred
+       ram with and without xbzrle
+       b. Migrate guest without heavy loaded inside guest and compare the
+       migration total time, downtime and transferred ram with and without
+       xbzrle
+    4) With xbzrle enabled, the total time, downtime and transferred ram
+       should less than disabled
+    5) Check live migration statistics for xbzrle specific options
+
+    :param test: kvm test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+
+    mig_protocol = params.get("mig_protocol", "tcp")
+    mig_type = utils_test.qemu.MultihostMigration
+    if mig_protocol == "fd":
+        mig_type = utils_test.qemu.MultihostMigrationFd
+    if mig_protocol == "exec":
+        mig_type = utils_test.qemu.MultihostMigrationExec
+    if "rdma" in mig_protocol:
+        mig_type = utils_test.qemu.MultihostMigrationRdma
+
+    class TestMultihostMigration(mig_type, MigrationBase):
+
+        """
+        multihost migration test
+        """
+
+        def __init__(self, test, params, env):
+
+            super(TestMultihostMigration, self).__init__(test, params, env)
+            self.srchost = self.params.get("hosts")[0]
```

Comment:
I think the two ways are all right. objects function gets hosts list by params.get("hosts").split(), use params.get("hosts")[0] can also get srchost.